Allow nvm to track latest Node.js LTS versions

colinrotherham opened the pull request:
This PR updates **.nvmrc** to use [LTS aliases `lts/*`](https://github.com/nvm-sh/nvm#long-term-support) rather than version numbers

### What's the problem?
We're seeing issues running `npm link` to add a locally-installed **govuk-frontend** into a new prototype

This is because Node.js version managers like [nvm](https://github.com/nvm-sh/nvm#readme) partition globally installed (and linked) packages, by Node version, into an `npm get prefix` prefixed directory:

```console
{prefix}/lib/node_modules/<package>
```

When preparing `govuk-frontend` for "linking" via [`npm link`](https://docs.npmjs.com/cli/v8/commands/npm-link#description) we will see a symbolic link added to one of many places, depending on the Node.js version set by **.nvmrc** config:

```console
/Users/username/.nvm/versions/node/v16.18.0/lib/node_modules/govuk-frontend
/Users/username/.nvm/versions/node/v16.14.2/lib/node_modules/govuk-frontend
/Users/username/.nvm/versions/node/v16.16.0/lib/node_modules/govuk-frontend
```

**`lts/gallium`** → https://github.com/alphagov/govuk-frontend/blob/main/.nvmrc
**`16.14.2`** → https://github.com/alphagov/govuk-design-system/blob/main/.nvmrc
**`16.16.0`** → https://github.com/alphagov/govuk-prototype-kit/blob/v13/.nvmrc

### What's the fix?

1. Temporarily align `.npmrc` in both projects each time we `npm link`
2. Or, ensure **.nvmrc** always uses [LTS aliases like `lts/gallium`](https://github.com/nvm-sh/nvm#long-term-support) not version numbers

**Note:** For 1) we'll see **EBADENGINE** without also setting `engine-strict=false` as we've enabled stricter checks

lfdebrux commented:
Hey Colin, thanks for the PR.

I'm a little confused about why the `.nvmrc` file is causing `EBADENGINE` warnings with `npm link`, I don't see why npm would care about what is in that file... can you elaborate more?

colinrotherham commented:
> Hey Colin, thanks for the PR.
> 
> I think this change makes sense, I'm just a little confused about why the `.nvmrc` file is causing `EBADENGINE` warnings with `npm link`, I don't see why npm would care about what is in that file... can you elaborate more?

No problem, updated the description @lfdebrux

lfdebrux commented:
Thanks for the more detailed description @colinrotherham. While I understand your use case, I am a little worried about the requirement for the Node version used by the Design System team and Prototype Kit team to be coupled. To be more specific, we're about to start recommending our users run the kit with Node 18, as that is soon going to be the current LTS; would you be happy to update the GOV.UK Frontend `.nvmrc` files to match?

colinrotherham commented:
@lfdebrux That sounds perfect

lfdebrux commented:
> @lfdebrux That sounds perfect

Cool. Thinking about it further, because for us new users will always be using the latest LTS, we should probably be doing the same, so maybe the `.nvmrc` file should just read `lts/*`. Would you be happy to update this PR to that?

colinrotherham commented:
@lfdebrux I like the idea, won't **package.json** `engines` (at some point) get out of sync and chuck warnings?

lfdebrux commented:
> @lfdebrux I like the idea, won't **package.json** `engines` (at some point) get out of sync and chuck warnings?

Yes, but we want that. If that happens to us it will happen to users too.

colinrotherham commented:
@lfdebrux I like that. Updated and pushed 👍 

I've removed the GitHub Action `node-version-file: .nvmrc` suggested change. Your workflow might pass on Node.js 18 but then fail when deployed to a Node.js 16 Heroku stack

colinrotherham commented:
Update now https://github.com/alphagov/govuk-frontend/pull/2939, https://github.com/alphagov/govuk-prototype-kit/pull/1712 and https://github.com/alphagov/govuk-design-system/pull/2399 are merged:

**`lts/gallium`** → https://github.com/alphagov/govuk-frontend/blob/main/.nvmrc
**`lts/gallium`** → https://github.com/alphagov/govuk-design-system/blob/main/.nvmrc
**`lts/*`** → https://github.com/alphagov/govuk-prototype-kit/blob/v13/.nvmrc

Once development environments are upgraded, **package.json** `engines` field changed, GitHub workflows bumped up, tests pass (and Heroku stacks upgraded to match) etc we may also want the `lts/*` alias everywhere
